### PR TITLE
Allow text-selection, but not copying, when `enablePermissions` is set (PR 16320 follow-up)

### PR DESCRIPTION
**Pull request by Snuffleupagus**
This implements part of https://github.com/mozilla/pdf.js/pull/16320#issuecomment-1514690567, however any UI-integration is ignored for now since as mentioned in https://github.com/mozilla/pdf.js/pull/16320#issuecomment-1514711525 that'd really benefit from UX feedback.

### CODE REVIEW

File: web/text_layer_builder.js [hunk 1]
```diff
@@ -215,11 +219,16 @@ class TextLayerBuilder {
...
 
     div.addEventListener("copy", event => {
-      const selection = document.getSelection();
-      event.clipboardData.setData(
-        "text/plain",
-        removeNullCharacters(normalizeUnicode(selection.toString()))
-      );
+      if (this.#enablePermissions) {
+        // Prevent copying.
+        event.clipboardData.setData("text/plain", "");
```
Comment (calixteman) [inline web/text_layer_builder.js:224]: Why are you overwriting clipboard content with an empty string ?
I checked in Acrobat and Chrome and the clipboard isn't changed.

Comment (Snuffleupagus) [reply web/text_layer_builder.js]: No good reason, so that's been fixed now :-)

**Review by calixteman — APPROVED** [review]
LGTM. Thank you.

**Pull request merged**